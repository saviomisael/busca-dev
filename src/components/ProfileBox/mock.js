import Image from '../../assets/images/person_white.svg';

export const profileBoxPropsMock = {
  avatar: Image,
  username: 'johndoe',
  fullname: 'John Doe',
  bio: 'bio Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.',
  blog: 'blog.com',
};
