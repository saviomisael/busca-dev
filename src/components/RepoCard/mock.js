export const repoCardPropsMock = {
  repoName: 'repo name',
  description: 'Repository Description lorem ipsum dolor sit amet, consectetur adipiscing elit.',
  starsCount: 0,
  repoUrl: 'github.com/repo',
};
