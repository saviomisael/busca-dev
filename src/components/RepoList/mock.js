export const repoListPropsMock = {
  repositories: [
    {
      repoId: 1,
      repoName: 'repo name',
      description: 'Repository Description lorem ipsum dolor sit amet, consectetur adipiscing elit.',
      starsCount: 0,
      repoUrl: 'github.com/repo',
    },
  ],
};

export const manyRepositories = {
  repositories: [
    {
      repoId: 1,
      repoName: 'repo name',
      description: 'Repository Description lorem ipsum dolor sit amet, consectetur adipiscing elit.',
      starsCount: 0,
      repoUrl: 'github.com/repo',
    },
    {
      repoId: 2,
      repoName: 'repo name',
      description: 'Repository Description lorem ipsum dolor sit amet, consectetur adipiscing elit.',
      starsCount: 3,
      repoUrl: 'github.com/repo',
    },
    {
      repoId: 3,
      repoName: 'repo name',
      description: 'Repository Description lorem ipsum dolor sit amet, consectetur adipiscing elit.',
      starsCount: 0,
      repoUrl: 'github.com/repo',
    },
    {
      repoId: 4,
      repoName: 'repo name',
      description: 'Repository Description lorem ipsum dolor sit amet, consectetur adipiscing elit.',
      starsCount: 5,
      repoUrl: 'github.com/repo',
    },
  ],
};
