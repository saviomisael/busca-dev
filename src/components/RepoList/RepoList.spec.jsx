import { screen } from '@testing-library/react';
import { RepoList } from '.';
import { repoListPropsMock, manyRepositories } from './mock';

describe('<RepoList />', () => {
  it('should renders list empty message when repositories list is empty', () => {
    global.renderTheme(<RepoList repositories={[]} />);

    const message = screen.getByText('Este usuário não possui repositórios');

    expect(message).toBeInTheDocument();
  });

  it('should renders a list of RepoCard when repositories length greater than zero', () => {
    global.renderTheme(<RepoList {...repoListPropsMock} />);

    const listElement = screen.getByRole('list');

    expect(listElement).toBeInTheDocument();
  });

  it('should renders many li elements', () => {
    global.renderTheme(<RepoList {...manyRepositories} />);

    const listItem = screen.getAllByRole('listitem');

    expect(listItem.length).toBe(4);
  });

  it('should matches with snapshot', () => {
    const { container } = global.renderTheme(<RepoList {...manyRepositories} />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
