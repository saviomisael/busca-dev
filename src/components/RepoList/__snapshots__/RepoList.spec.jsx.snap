// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RepoList /> should matches with snapshot 1`] = `
.c0 {
  padding: 2rem;
  border-radius: 0.6rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  row-gap: 2rem;
  background: linear-gradient(#D9D9D9BB,#353535CC);
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-column-gap: 0.7rem;
  column-gap: 0.7rem;
}

.c7 {
  text-transform: uppercase;
  font-size: 1.6rem;
  color: #284B63;
}

.c6 {
  -webkit-filter: invert(23%) sepia(6%) saturate(4839%) hue-rotate(162deg) brightness(93%) contrast(82%);
  filter: invert(23%) sepia(6%) saturate(4839%) hue-rotate(162deg) brightness(93%) contrast(82%);
}

.c1 {
  padding: 1.5rem;
  border-radius: 0.6rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  row-gap: 1rem;
  background-color: #D9D9D9;
}

.c1 > * {
  font-size: 1.8rem;
  color: #353535;
}

.c1 > .c4 {
  margin-top: 1rem;
}

.c2 {
  font-weight: 600;
}

.c3 {
  font-weight: 500;
}

<ul
  class="c0"
>
  <li
    class=""
  >
    <article
      class="c1"
    >
      <a
        class="c2"
        href="github.com/repo"
        target="_blank"
      >
        repo name
      </a>
      <p
        class="c3"
      >
        Repository Description lorem ipsum dolor sit amet, consectetur adipiscing elit.
      </p>
      <div
        class="c4 c5"
      >
        <img
          alt="star icon"
          class="c6"
          src="star_empty.svg"
        />
        <p
          class="c7"
        >
          0
           Estrelas
        </p>
      </div>
    </article>
  </li>
  <li
    class=""
  >
    <article
      class="c1"
    >
      <a
        class="c2"
        href="github.com/repo"
        target="_blank"
      >
        repo name
      </a>
      <p
        class="c3"
      >
        Repository Description lorem ipsum dolor sit amet, consectetur adipiscing elit.
      </p>
      <div
        class="c4 c5"
      >
        <img
          alt="star icon"
          class="c6"
          src="star_fill.svg"
        />
        <p
          class="c7"
        >
          3
           Estrelas
        </p>
      </div>
    </article>
  </li>
  <li
    class=""
  >
    <article
      class="c1"
    >
      <a
        class="c2"
        href="github.com/repo"
        target="_blank"
      >
        repo name
      </a>
      <p
        class="c3"
      >
        Repository Description lorem ipsum dolor sit amet, consectetur adipiscing elit.
      </p>
      <div
        class="c4 c5"
      >
        <img
          alt="star icon"
          class="c6"
          src="star_empty.svg"
        />
        <p
          class="c7"
        >
          0
           Estrelas
        </p>
      </div>
    </article>
  </li>
  <li
    class=""
  >
    <article
      class="c1"
    >
      <a
        class="c2"
        href="github.com/repo"
        target="_blank"
      >
        repo name
      </a>
      <p
        class="c3"
      >
        Repository Description lorem ipsum dolor sit amet, consectetur adipiscing elit.
      </p>
      <div
        class="c4 c5"
      >
        <img
          alt="star icon"
          class="c6"
          src="star_fill.svg"
        />
        <p
          class="c7"
        >
          5
           Estrelas
        </p>
      </div>
    </article>
  </li>
</ul>
`;
